include/stl2/detail/memory/destroy.hpp

```diff
@@ -60,10 +50,9 @@ STL2_OPEN_NAMESPACE {
 		return first;
 	}
 
-	template <InputRange Rng>
+	template <__NoThrowInputRange Rng>
 	requires
-		Destructible<value_type_t<iterator_t<Rng>>> &&
-		__ReferenceTo<iterator_t<Rng>, value_type_t<iterator_t<Rng>>>
+		Destructible<value_type_t<iterator_t<Rng>>>
 	safe_iterator_t<Rng> destroy(Rng&& rng) noexcept
```

Comment:
Need to document that `begin`/`end` are required to be nothrow somewhere.

Reply:
I'm torn between requiring this as a part of `__NoThrowInputRange` and requiring it right here. I'd prefer it be in the concept, but I'm concerned that will over-constrain `uninitialised_foo`.